nanoFramework.Device.Bluetooth/GenericAttributeProfile/GattLocalCharacteristic.cs

```diff
@@ -0,0 +1,400 @@
+                writer.WriteBuffer(_staticValue);
+                e.GetRequest().RespondWithValue(_staticValue);
+            }
+            else if (descritorId != 0)
+            {
+                // Descriptor event, let descriptor handle it
+                GattLocalDescriptor des = FindDescriptor(descritorId);
+                if (des != null)
+                {
+                    handled = des.OnReadRequested(e);
+                }
+            }
+            else if (ReadRequested != null)
+            {
+                // LocalCharacteristic event ?
+                handled = true;
+                ReadRequested?.Invoke(this, e);
+            }
+
+            if (!handled)
+            {
+                // No event handler, respond with error
+                e.GetRequest().RespondWithProtocolError(GattProtocolError.UnlikelyError);
+            }
+        }
+
+        internal void OnWriteRequested(UInt16 descritorId, GattWriteRequestedEventArgs e)
+        {
+            bool handled = false;
+
+            if (WriteRequested != null)
+            {
+                // LocalCharacteristic event ?
+                if (descritorId == 0)
+                {
+                    handled = true;
+                    WriteRequested?.Invoke(this, e);
+                }
+                else
+                {
+                    // Descriptor event
+                    GattLocalDescriptor des = FindDescriptor(descritorId);
+                    if (des != null)
+                    {
+                        handled = des.OnWriteRequested(e);
+                    }
+                }
+            }
+
+            if (!handled)
+            {
+                // No event handler, respond with error
+                e.GetRequest().RespondWithProtocolError(GattProtocolError.UnlikelyError);
+            }
+        }
+
+        private int ClientSubscribed(GattSubscribedClient client)
+        {
+            for(int index=0; index<_subscribedClients.Count; index++)
+            {
+                if ( ((GattSubscribedClient)_subscribedClients[index]).Session.DeviceId.Id == client.Session.DeviceId.Id)
+                {
+                    return index;
+                }
+            }
+            return -1;
+        }
+
+        internal void OnSubscribedClientsChanged(bool subscribe, GattSubscribedClient client)
+        {
+            lock (_subscribedClients)
+            {
+                int index = ClientSubscribed(client);
+
+                if (subscribe)
+                {
+                    if (index < 0)
+                    {
+                        _subscribedClients.Add(client);
+                    }
+                }
+                else
+                {
+                    if (index >= 0)
+                    {
+                        _subscribedClients.RemoveAt(index);
+                    }
+                }
+            }
+            SubscribedClientsChanged?.Invoke(this, new object());
```

Comment:
is it necessary to create an object, won't null be enough?

Reply:
Yes null should be ok